packages/collector/test/tracing/cloud/azure/blob/test.js

```diff
@@ -0,0 +1,477 @@
+          dataProperty: 'azstorage',
+          op: 'download',
+          n: 1,
+          spans: spans
+        });
+        await verify({
+          spanName: 'azstorage',
+          dataProperty: 'azstorage',
+          n: 1,
+          path: '/download-promise-err',
+          withError: true,
+          spans: spans
+        });
+      });
+    });
+
+    it('download - err', async () => {
+      await controls.sendRequest({
+        method: 'GET',
+        path: '/download-err'
+      });
+
+      await testUtils.retry(async () => {
+        const spans = await agentControls.getSpans();
+        await verifyOp({
+          dataProperty: 'azstorage',
+          op: 'download',
+          n: 1,
+          spans: spans
+        });
+        await verify({
+          spanName: 'azstorage',
+          dataProperty: 'azstorage',
+          n: 1,
+          path: '/download-err',
+          withError: true,
+          spans: spans
+        });
+      });
+    });
+
+    it('download-blockblob-promise', async () => {
+      await controls.sendRequest({
+        method: 'GET',
+        path: '/download-blockblob-promise'
+      });
+
+      await testUtils.retry(async () => {
+        const spans = await agentControls.getSpans();
+        await verifyOp({
+          dataProperty: 'azstorage',
+          op: 'download',
+          n: 1,
+          spans: spans
+        });
+        await verify({
+          spanName: 'azstorage',
+          dataProperty: 'azstorage',
+          n: 3,
+          path: '/download-blockblob-promise',
+          withError: false,
+          spans: spans
+        });
+      });
+    });
+
+    async function verify({ spanName, dataProperty, n, path, withError, spans }) {
+      const _pid = String(controls.getPid());
+      const parent = verifyHttpRootEntry({
+        spans,
+        apiPath: path,
+        pid: _pid
+      });
+      return expectExactlyNMatching(spans, n, [
+        span => expect(span.n).to.equal('azstorage'),
+        span => expect(span.k).to.equal(constants.EXIT),
+        span => expect(span.t).to.equal(parent.t),
+        span => expect(span.p).to.equal(parent.s),
+        span => expect(span.f.e).to.equal(_pid),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.ec).to.equal(withError ? 1 : 0),
+        span => expect(span.data).to.exist,
+        span =>
+          withError ? expect(span.data[spanName].error).to.exist : expect(span.data[spanName].error).to.be.undefined,
+        span => expect(span.data[dataProperty || spanName]).to.be.an('object'),
+        span => expect(span.data[dataProperty || spanName].accountName).to.exist,
+        span => expect(span.data[dataProperty || spanName].blobName).to.exist,
+        span => expect(span.data[dataProperty || spanName].containerName).to.exist,
+        span => expect(span.data[dataProperty || spanName].op).to.exist
+      ]);
+    }
+    async function verifyOp({ dataProperty, op, n, spans }) {
```

Comment:
I still don't understand why we need two verify functions. You could check the operation in the verify function.

Reply:
resolved